qpcr_analysis.py

```diff
@@ -292,14 +300,19 @@
     df_Ct = data[['Cq', 'Sample names']].copy()
     df_Ct = df_Ct.rename(columns={'Cq' : 'Ct value', 'Sample names' : 'Sample'})
 
-    # Remove Mq/H2O samples, as those are empty
-    for i in range(0,df_Ct.shape[0]):
-        if df_Ct.loc[i,'Sample'].partition('_')[0] == 'mq':
-            df_Ct = df_Ct.drop([i])
-    df_Ct = df_Ct.reset_index(drop=True)
-    unique_cell_lines = unique_cell_lines[unique_cell_lines !='mq']
-    unique_cell_lines = unique_cell_lines[unique_cell_lines !='MQ']
-    nr_samples = nr_samples - 1
+# Remove Mq/H2O samples, as those are empty
+for i in range(0,df_Ct.shape[0]):
+    if df_Ct.loc[i,'Sample'].partition('_')[0] == 'h2o':
+        df_Ct = df_Ct.drop([i])
+    elif df_Ct.loc[i,'Sample'].partition('_')[0] == 'H2O':
+        df_Ct = df_Ct.drop([i])
+    elif df_Ct.loc[i,'Sample'].partition('_')[0] == 'mq':
+        df_Ct = df_Ct.drop([i])
+    elif df_Ct.loc[i,'Sample'].partition('_')[0] == 'MQ':
+        df_Ct = df_Ct.drop([i])
+df_Ct = df_Ct.reset_index(drop=True)
+unique_cell_lines = unique_cell_lines_MQ_removed
+nr_samples = nr_samples - 1

```

Comment:
It's better not just to decrease the number of `nr_samples`, but to recalculate it:
```python
nr_samples = len(unique_cell_lines)
```